java-frontend/src/main/java/org/sonar/java/resolve/Types.java

```diff
@@ -194,4 +246,135 @@ public int compare(Type type, Type t1) {
     return results;
   }
 
+  /**
+   * For any element G of MEC that is a generic type, let the "relevant" parameterizations of G, Relevant(G), be:
+   * Relevant(G) = { V | 1 ≤ i ≤ k: V in ST(Ui) and V = G<...> }
+   *
+   * @param minimalErasedCandidates MEC
+   * @param supertypes
+   * @return the set of known parameterizations for each generic type G of MEC 
+   */
+  private static Multimap<Type, Type> relevantParameterizations(List<Type> minimalErasedCandidates, List<Set<Type>> supertypes) {
+    Multimap<Type, Type> result = HashMultimap.create();
+    for (Set<Type> supertypesSet : supertypes) {
+      for (Type supertype : supertypesSet) {
+        Type erasedSupertype = supertype.erasure();
+        if (minimalErasedCandidates.contains(erasedSupertype)) {
+          result.put(erasedSupertype, supertype);
+        }
+      }
+    }
+    return result;
+  }
+
+  @VisibleForTesting
+  static Type best(List<Type> minimalCandidates) {
+    Collections.sort(minimalCandidates, new Comparator<Type>() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1604)